### PR TITLE
clarified the value of the 'type' node

### LINKED ISSUE #358: Clarify the value of the 'type' node
The value of a `type` node must be the name of either a user-defined type or one of the built-in types (object, array, or a scalar type), or a JSON or XML Schema.

### PR DESCRIPTION
**Pull request by sichvoge**
closes #358

### CODE REVIEW

File: versions/raml-10/raml-10.md [hunk 1]
```diff
@@ -434,7 +434,7 @@ declarations may have; certain type declarations may have other properties.
 |:----------|:----------|
 | default? | Provides a default value for a type.
 | schema? | Alias for the equivalent "type" property, for compatibility with RAML 0.8. Deprecated - API definitions should use the "type" property, as the "schema" alias for that property name may be removed in a future RAML version. The "type" property allows for XML and JSON schemas.
-| type? | A base type which the current type extends, or more generally a type expression.
+| type? | A base type which the current type extends, or more generally a type expression. The value of a type node must be the name of either a user-defined type or one of the built-in types (object, array, or a scalar type), or a JSON or XML Schema.
```
Comment (usarid) [inline versions/raml-10/raml-10.md]: To be perfectly clear, it MUST be either a) the name of a user-defined RAML data type or b) the name of a built-in RAML data type (object, array, or scalar type) or c) a RAML data type expression that uses the names of user-defined or built-in RAML data types (e.g. `User[] | string[]`) or d) the name of a JSON or XML schema.

Comment (sichvoge) [inline versions/raml-10/raml-10.md]: That's not true isn't it? It can also be an inline JSON or XML Schema right?

Comment (usarid) [inline versions/raml-10/raml-10.md]: If so, then why would we not allow an inline RAML data type declaration? I thought that for various good reasons we chose to not allow inline schema definitions of any sort. Let's take a look again at why we did that and see why we might want to treat external schemas differently.

**Comment by sichvoge** [comment]
See new commit.

File: versions/raml-10/raml-10.md [hunk 2]
```diff
@@ -434,7 +434,7 @@ declarations may have; certain type declarations may have other properties.
 |:----------|:----------|
 | default? | Provides a default value for a type.
 | schema? | Alias for the equivalent "type" property, for compatibility with RAML 0.8. Deprecated - API definitions should use the "type" property, as the "schema" alias for that property name may be removed in a future RAML version. The "type" property allows for XML and JSON schemas.
-| type? | A base type which the current type extends, or more generally a type expression.
+| type? | A base type which the current type extends, or more generally a type expression. The value of a `type` node MUST be either a) the name of a user-defined type or b) the name of a built-in RAML data type (object, array, or scalar type) or c) an inline type declaration.
```
Comment (usarid) [inline versions/raml-10/raml-10.md]: Change

>A base type which the current type extends, or more generally a type expression. The value of a `type` node MUST be either a) the name of a user-defined type or b) the name of a built-in RAML data type (object, array, or scalar type) or c) an inline type declaration.

to

>A base type which the current type extends or just wraps. The value of a `type` node MUST be either a) the name of a user-defined type or b) the name of a built-in RAML data type (object, array, or one of the scalar types) or c) an inline type declaration.

**Comment by usarid** [comment]
See my minor changes, else LGTM

**Pull request merged**